src/details/ArborX_DetailsTreeTraversal.hpp

```diff
@@ -505,9 +505,10 @@ struct TreeTraversal<BVH, Predicates, Callback,
         if (!heap.empty() && heap.top().second < closer_pair.second)
         {
           node = heap.top().first;
-          heap.pop();
           if (closer_pair.second < inf)
-            heap.push(closer_pair);
```

Comment:
I'm pretty sure I tried this and found it to be slower for the benchmark, also see https://github.com/arborx/ArborX/pull/683#discussion_r888088051 and https://github.com/arborx/ArborX/pull/683#discussion_r891529247.

Reply:
Regardless of what we decide, this will be worth a comment in code

Reply:
Which benchmark did you use? I couldn't find anything in `benchmark`. I tried to use the ray tracing example but the results are not stable enough on HIP.  I get more than 25% difference between consecutive runs.

Reply:
Yeah, I think the ray tracing example is the only one we can use for performance testing.

The variation is interesting, and is a bit unexpected. Wonder if that's also the case for the CUDA build. Could it be a problem size issue? If you are running with the default values it is a really small problem.

Reply:
I was using HIP on crusher. I tried to change some of the input parameters but it didn't help